### PR TITLE
Plumb `context.Context` down into functions that use `util/cmd` functions

### PR DESCRIPTION
**Pull request by MarlonGamez**
Related: #5368 

**Description**
This PR plumbs `context.Context` parameter into functions in order to have a `context.Context` variable in `util.RunCmd` and related functions

### CODE REVIEW

File: hack/versions/pkg/schema/git.go [hunk 1]
```diff
@@ -73,7 +74,7 @@ func (g *git) diffWithBaseline(path string) ([]byte, error) {
 
 func (g *git) run(args ...string) ([]byte, error) {
 	cmd := exec.Command(g.path, args...)
-	out, err := util.RunCmdOut(cmd)
+	out, err := util.RunCmdOut(context.Background(), cmd)
```
Comment (tejal29) [inline hack/versions/pkg/schema/git.go:77]: There is another PR to move replace context.Background to context.TODO.
Do you want to sync with that and see why `TODO` is preferred?

From docs, looks like Background context is never cancelled. 
https://pkg.go.dev/context#Background

**Review by tejal29 — APPROVED** [review]

**Comment by tejal29** [comment]
Code changes Look good. 
Trying this out.

**Comment by tejal29** [comment]
After applying a diff and some validation I can now see debug logs in relevant nodes. 
e.g. skaffold deploy logs are not seen in deploy node for `skaffold` run config with debug level logging.

image



Diff i applied:
```
diff --git a/pkg/skaffold/event/v2/logger.go b/pkg/skaffold/event/v2/logger.go
index d38e68a8c..bf19f8e9c 100644
--- a/pkg/skaffold/event/v2/logger.go
+++ b/pkg/skaffold/event/v2/logger.go
@@ -86,9 +86,16 @@ func (h logHook) Levels() []logrus.Level {
 
 // Fire constructs a SkaffoldLogEvent and sends it to the event channel
 func (h logHook) Fire(entry *logrus.Entry) error {
+       task := constants.DevLoop
+       if s, ok := entry.Data["task"]; ok {
+               fmt.Println("found for entry", entry.Message)
+               task = s.(constants.Phase)
+       } else {
+               fmt.Println("not found for ", entry.Message, entry.Data)
+       }
        handler.handleSkaffoldLogEvent(&proto.SkaffoldLogEvent{
-               TaskId:    fmt.Sprintf("%s-%d", h.task, handler.iteration),
-               SubtaskId: h.subtask,
+               TaskId:    fmt.Sprintf("%s-%d", task, handler.iteration),
+               SubtaskId: entry.Data["subtask"].(string),
                Level:     levelFromEntry(entry),
                Message:   entry.Message,
        })
```

**Comment by tejal29** [comment]
Testing for BUILD scenario. 

Seeing `mvn build` command in the build node for frontend and backend.

image

**Comment by tejal29** [comment]
Portfowarding logs are in the right section
image

**Comment by tejal29** [comment]
Follow up cleaning tasks
1) `output.WithContext` should also return the `traceContext` and `endTrace` or `instrumentation.StartTrace` should preserve context values.
right now, `instrumentation.StartTrace(ctx, "Cleanup")` clears all context values if tracing if enabled.

File: pkg/skaffold/event/v2/logger.go [hunk 2]
```diff
@@ -87,8 +81,8 @@ func (h logHook) Levels() []logrus.Level {
 // Fire constructs a SkaffoldLogEvent and sends it to the event channel
 func (h logHook) Fire(entry *logrus.Entry) error {
 	handler.handleSkaffoldLogEvent(&proto.SkaffoldLogEvent{
-		TaskId:    fmt.Sprintf("%s-%d", h.task, handler.iteration),
-		SubtaskId: h.subtask,
+		TaskId:    fmt.Sprintf("%s-%d", entry.Data["task"], handler.iteration),
```
Comment (tejal29) [inline pkg/skaffold/event/v2/logger.go:84]: Can you verify if this will not cause a null pointer exception

Comment (tejal29) [reply pkg/skaffold/event/v2/logger.go:84]: (if the key does not exists)

Comment (MarlonGamez) [reply pkg/skaffold/event/v2/logger.go]: my bad, just pushed changes to fix this

**Review by tejal29 — APPROVED** [review]
LGTM for Monday in advance.

**Comment by gsquared94** [comment]
can this be merged @MarlonGamez ?

File: hack/versions/pkg/schema/git.go (hunk 1 above)
Comment (gsquared94) [reply hack/versions/pkg/schema/git.go:77]: I think that `context.TODO` is a hint to the dev that if ever a proper context becomes available in future by a change in the function parameters or something, then plumb that through. Whereas `context.Background` is a deliberate use even if a parent context is available. That's what I understood from @ahmetb's PR.

Comment (ahmetb) [reply hack/versions/pkg/schema/git.go:77]: yeah that's the case. ideally a context.Background() is for a call stack that can be cancelled from a central place, e.g. Ctrl-C. Background threads with separate lifecycle (as the name also says) get a new Background context.

If you don't plumb yet but intend do plumb someday, you can use context.TODO, which is identical to Background() functionally, only different in semantics and indicates your intention.

**Pull request merged**